### PR TITLE
Refactor reactions

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-406

### Description

- Refactored reaction views
- Made a separate reaction view specifically for a list of reactions made by particular users
- Improved the performance of `ViewReactionsView` in the message list 
- Removed unnecessary spacing between a message and `EditReactionsView` on the message options overlay

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsView.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            setPadding(it, 0, it, 0)
+        }
+
+        adapter = ReactionsAdapter(reactionsViewStyle) {
+            reactionClickListener?.onReactionClick(it)
+        }.also { reactionsAdapter = it }
+    }
+
+    private fun createReactionItems(message: Message): List<ReactionItem> {
+        val reactionsMap = mutableMapOf<String, ReactionItem>()
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsView.kt:82]: You're using only the `values` of the map. A `Set` looks better for this case.

**Review by zsmb13 — COMMENTED** [review]
Good job, found just one significant thing to fix up around attributes

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsBubbleDrawer.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        } else {
+            if (isSingleReaction) {
+                bubbleWidth / 2 - bubbleOffset
+            } else {
+                MULTIPLE_REACTIONS_BASELINE_OFFSET - bubbleOffset
+            }
+        }
+    }
+
+    private companion object {
```
[the remark this answers is not in the document]
Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsBubbleDrawer.kt:115]: Took a look, we don't.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsView.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/view/ViewReactionsView.kt:82]: But I'm using `containsKey(reaction.type)` and `reactionsMap[reaction.type]`, maybe I'm missing something?

**Review by zsmb13 — APPROVED** [review]
Looks good as far as my comments go 👍

**Pull request merged**